### ISSUE TITLE
Logistic Regression Issue without "eval()"

### ISSUE THREAD

**Issue by jaeohwoo** [Spijkervet/SimCLR#19]
At https://github.com/Spijkervet/SimCLR/blob/master/testing/logistic_regression.py,

simclr_model needs to be set "eval" mode.

At line 168:
    # load pre-trained model from checkpoint
    simclr_model = SimCLR(args, encoder, n_features)
    **_simclr_model.eval()_** # need to add this line
    model_fp = os.path.join(
        args.model_path, "checkpoint_{}.tar".format(args.epoch_num)
    )
    simclr_model.load_state_dict(torch.load(model_fp, map_location=args.device.type))
    simclr_model = simclr_model.to(args.device)

Without "eval" mode, the simclr network weights were unintentionally updated when we extract features (especially for testing set). Then it significantly reduces the testing accuracy for my test cases. 

We need to add ".eval()". Please check it out and confirm.

Thank you.

**Comment by Spijkervet** [comment]
Hi Jaeohwoo,

Thank you for submitting your issue. I've immediately reviewed the code and ran a test with the line `simclr_model.eval()` added to the model initialisation. This did not result in a drop in accuracy for me.

The following happens in `testing/logistic_regression.py`:
1. All data is forward-passed through a pre-trained (frozen!) SimCLR network.
2. These representations (**h**) are saved in a giant tensor in memory.
3. We train a linear logistic regression model using these representations.

I have gone ahead and added the eval() in [this](https://github.com/Spijkervet/SimCLR/commit/654f05f107ce17c0a9db385f298a2dc6f8b3b870) commit.
All best,

Janne
